src/components/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,142 @@
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting, isValid },
+  } = useForm<FormData>({
+    resolver: zodResolver(schema),
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+  });
+
+  const onSubmit = async (data: FormData) => {
+    try {
+      const raw = localStorage.getItem('users');
+      const users: User[] = raw ? JSON.parse(raw) : [];
+
+      if (users.some(user => user.email === data.email)) {
+        throw new Error('Email already registered');
+      }
+
+      const newUser: User = {
+        id: uuidv4(),
+        fullName: data.fullName,
+        email: data.email,
+        password: data.password,
+      };
+
+      users.push(newUser);
+      localStorage.setItem('users', JSON.stringify(users));
+
+      navigate('/signin');
+    } catch (err: unknown) {
+      if (err instanceof Error) {
+        alert(err.message);
+      } else {
+        alert('An unknown error occurred');
+      }
+    }
+  };
+
+  return (
+    <Container maxWidth="sm">
+      <Box
+        sx={{
+          display: 'flex',
+          justifyContent: 'center',
+          mt: 8,
+        }}
+      >
+        <Paper elevation={4} sx={{ p: 4, width: '100%' }}>
+          <Typography variant="h4" align="center" gutterBottom>
+            Create Account
+          </Typography>
+
+          <Box component="form" onSubmit={handleSubmit(onSubmit)} noValidate>
+            <Stack spacing={2}>
+              <TextField
+                label="Full Name"
+                fullWidth
+                autoFocus
+                {...register('fullName')}
+                error={!!errors.fullName}
+                helperText={errors.fullName?.message}
+              />
+
+              <TextField
+                label="Email"
+                type="email"
+                fullWidth
+                {...register('email')}
+                error={!!errors.email}
+                helperText={errors.email?.message}
+              />
+
+              <TextField
+                label="Password"
+                type="password"
+                fullWidth
+                {...register('password')}
+                error={!!errors.password}
+                helperText={errors.password?.message}
+              />
+
+              <TextField
+                label="Confirm Password"
+                type="password"
+                fullWidth
+                {...register('confirm')}
+                error={!!errors.confirm}
+                helperText={errors.confirm?.message}
+              />
+
+              <Button
+                type="submit"
+                variant="contained"
+                size="large"
+                fullWidth
+                disabled={!isValid || isSubmitting}
+                sx={{ mt: 1 }}
+              >
+                Sign Up
+              </Button>
+            </Stack>
+          </Box>
+
+          <Typography variant="body2" align="center" sx={{ mt: 3 }}>
+            Already have an account?{' '}
```

Comment:
Remove this extra space
```suggestion
            Already have an account?
```

Reply:
In JSX, {''} is used to insert a space between elements. Without it, they would appear as a continuous string. so to ensures proper spacing and readability I used it

Reply:
I don't think it will be, try it out.

Reply:
I tried it problem is occurred as i mentioned above .